### PR TITLE
Input Validation for N in Sim 1

### PR DESCRIPTION
**Pull request by Evan-CTL**

### CODE REVIEW

**Comment by zmustapha** [comment]
I could still type any number larger than 500

**Review by ndittren — DISMISSED** [review]
I think this can be better achieved by taking advantage of HTML event changes. We can add another state that will be handled by the onChange and make the necessary validation with an onBlur. The handleNBlur function would be the one to make the validation checks. These changes can be done just in simulationOne.jsx

**Comment by Evan-CTL** [comment]
> I think this can be better achieved by taking advantage of HTML event changes. We can add another state that will be handled by the onChange and make the necessary validation with an onBlur. The handleNBlur function would be the one to make the validation checks. These changes can be done just in simulationOne.jsx

Agreed. Changes made

**Review by ndittren — CHANGES_REQUESTED** [review]
Address comment left on simulationOne.jsx line 54

File: media/js/src/simulationOne/simulationOne.jsx [hunk 1]
```diff
@@ -51,7 +51,16 @@ export const SimulationOne = () => {
     };
 
     const handleNChange = (e) => {
-        setN(parseInt(e.target.value));
+        setN(parseInt(e.target.value) || N);
```
[the remark this answers is not in the document]
Comment (Evan-CTL) [reply media/js/src/simulationOne/simulationOne.jsx:54]: Oh, that was what the review was hanging on. It didn't pop up in the conversation. My mistake.

**Review by ndittren — APPROVED** [review]

**Pull request merged**